### PR TITLE
@craigspaeth => auction2 add handling for confirm-registration modal

### PR DESCRIPTION
**Pull request by sweir27**
Couple of minor changes here, plus some test additions (since we didn't have any for routing!).

- Intentionally gets rid of the "Notify me when the auction starts" form field that previously showed up for non-logged in users on sales in preview mode. Instead, we'll just show the register to bid button and it will first ask them to log in (this is the behavior once the auction starts anyways).
image

- Adds handling for the `/confirm-registration` modal that pops up on a redirect after registration. Note that I didn't hook up the registration page to redirect to `/auction2`, but if you visit `/auction2/:id/confirm-registration` you can view the modal as expected.
image

- Added a bunch of tests... I'm still looking into why the last few router tests are failing 😞 

Open questions:
- What does this redirect do? Do we need to keep it? https://github.com/artsy/force/blob/master/desktop/apps/auction/routes.coffee#L90-L97
- I can't find any references to `feature`s on the auction page, but do you happen to know of a case where they are used? https://github.com/artsy/force/blob/master/desktop/apps/auction/client/index.coffee#L17

### CODE REVIEW

File: desktop/apps/auction2/test/routes.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            id: 'foo'
+          }
+        }]
+      }
+    }
+
+    RoutesRewireApi.__Rewire__('metaphysics', sinon.stub().returns(Promise.resolve(auctionQueries)))
+    routes.redirectLive(req, res, next)
+    res.redirect.called.should.not.be.ok()
+    next.called.should.be.ok()
```
Comment (sweir27) [inline desktop/apps/auction2/test/routes.js]: Am I doing something obviously wrong here with my `sinon` invocation? I'm passing `next` into the router function and it's being called there, but this line fails. 😫

File: desktop/apps/auction2/routes.js [hunk 2]
```diff
@@ -11,3 +12,18 @@ export const index = async (req, res) => {
...
+
+export const redirectLive = async (req, res, next) => {
+  const { sale } = await metaphysics({ query: SaleQuery(req.params.id) })
+  if (sale && sale.is_live_open) {
+    const liveUrl = `${PREDICTION_URL}/${sale.id}/login`
+    const { me } = await metaphysics({ query: MeQuery(req.params.id), req: req })
+    if (me && me.bidders && me.bidders.length > 0 && me.bidders[0].qualified_for_bidding) {
+      res.redirect(liveUrl)
+    } else {
+      return next()
```
Comment (craigspaeth) [inline desktop/apps/auction2/routes.js]: These returns are probably not necessary (assuming they were generated from a coffeescript conversion).

File: desktop/apps/auction2/test/routes.js (hunk 1 above)
Comment (craigspaeth) [inline desktop/apps/auction2/test/routes.js]: `redirectLive` is an async function—which means it returns a promise. Even if you stub out `metaphysics` to resolve immediately, the wrapping promise will still defer execution to the next event loop tick.

So my hunch is you simply need to tag the `it` callback async and await `redirectLive`

````javascript
it('does not redirect if bidder is not qualified', async () => {
  // ...
  await routes.redirectLive(req, res, next)
  next.called.should.be.ok()
})
````

File: desktop/apps/auction2/routes.js (hunk 2 above)
Comment (sweir27) [inline desktop/apps/auction2/routes.js]: Ah, I think I added these in to try to fix my test issue :)

File: desktop/apps/auction2/test/routes.js (hunk 1 above)
Comment (sweir27) [inline desktop/apps/auction2/test/routes.js]: OH! Thank you for the explanation! 🙌 That works like a charm.

**Comment by craigspaeth** [comment]
> What does this redirect do? Do we need to keep it?
> https://github.com/artsy/force/blob/master/desktop/apps/auction/routes.coffee#L90-L97

Auction pages used to be hobbled together by Gravity's feature model (a weird frankenstein model for piecing together static-ish pages of ordered sets and featured links). I believe this was a defensive check for old auction urls that were built as these features. Let's leave it out now—I don't imagine we'll need to support this anymore as it's at-least a couple years since we built auctions like that. 

> I can't find any references to features on the auction page, but do you happen to know of a case where they are used?
> https://github.com/artsy/force/blob/master/desktop/apps/auction/client/index.coffee#L17

Echoing above—it was an old way of building auction pages that we should try to leave buried away in our nightmares.

**Comment by craigspaeth** [comment]
👏  looks good! No blockers on my end—feel free to merge when 💚 

Side note: Looking at [client.js](https://github.com/sweir27/force/blob/5322adfdca9eab435859b0551f2b8ae3c0ec0e81/desktop/apps/auction2/client.js) I can see how mixing the jQuery/Backbone way with React can get a little disorderly. I wonder if we could start exporting a React wrapper to our Backboney components (e.g. using lifecycle methods to setup/teardown) so these can more cleanly integrate in a React component tree.

**Comment by damassi** [comment]
@craigspaeth - i started thinking about just that thing a few weeks back when I first started exploring force and sketched something out https://github.com/damassi/backbone-react-view. Not extensively tested however.

**Comment by sweir27** [comment]
Yeah, would totally like to jam on making the `client.js` cleaner! (and I'll merge on 🍏 ! Just trying to get the circle.yml to behave)

**Comment by craigspaeth** [comment]
@damassi Nice! Yeah I remember looking at that 👍 . My hunch is, unfortunately, a generic abstraction probably won't work given component design doesn't adhere to any formal API conventions (e.g. some don't even use Backbone views and just export a function that applies jQuery stuff). So we'd probably end up having to write a new wrapper for each component as we go.

**Comment by damassi** [comment]
Yeah, that was my thinking too -- that as it scales to all of the different component implementations in Force so many edge-cases would appear that would basically complicate things further.

**Pull request merged**